### ISSUE TITLE
Zeros left on imei (identifier)

### ISSUE THREAD

**Issue by glbatistabh** [traccar/traccar#3070]
Leading zeros in the identifier (imei)

I believe there may be some problem with the traccar.

I'm using the GT02A GPS (port 5002) and traccar it with the "auto registration" setting enabled.

```
    <entry key='database.registerUnknown'>true</entry>
    <entry key='database.registerUnknown.defaultGroupId'>1</entry>
    <entry key='database.registerUnknown.defaultCategory'>Default</entry>
```

Sometimes the tracer recognizes the identifier "000027042385393" and at other times it recognizes the identifier 027042385393 (without 3 leading zeros). I think it may be somehow a formatting problem for the identifier.

See the attached log file.
[stonetech.info/tracker-server.log.20170401](url)

```
2017-04-01 12:33:40  INFO: [61CA5735] id: 000027042385393, time: 2017-04-01 12:33:39, lat: -19.90649, lon: -43.97801, speed: 1.4, course: 144.5
2017-04-01 12:34:10  INFO: [61CA5735] id: 000027042385393, time: 2017-04-01 12:34:09, lat: -19.90646, lon: -43.97722, speed: 3.5, course: 79.6
2017-04-01 13:43:03  INFO: Position filtered by Duplicate Distance filters from device: 000027042385393 with id: 1
2017-04-01 13:44:02  INFO: [42E896BE] id: 000027042385393, time: 2017-04-01 13:44:00, lat: -19.90388, lon: -43.97727, speed: 7.3, course: 356.5
2017-04-01 13:44:31  INFO: [42E896BE] id: 000027042385393, time: 2017-04-01 13:44:30, lat: -19.90423, lon: -43.97656, speed: 9.4, course: 161.3
2017-04-01 13:45:02  INFO: [42E896BE] id: 000027042385393, time: 2017-04-01 13:45:00, lat: -19.90506, lon: -43.97664, speed: 0.8, course: 197.3
2017-04-01 13:45:32  INFO: Position filtered by Distance filters from device: 000027042385393 with id: 1
2017-04-01 13:54:56  INFO: Position filtered by Distance filters from device: 000027042385393 with id: 1
2017-04-01 13:55:31  INFO: [F3DB58C0] id: 000027042385393, time: 2017-04-01 13:55:30, lat: -19.90641, lon: -43.97664, speed: 0.8, course: 226.5
2017-04-01 13:56:01  INFO: [F3DB58C0] id: 000027042385393, time: 2017-04-01 13:55:59, lat: -19.90818, lon: -43.97670, speed: 7.8, course: 191.4
2017-04-01 13:56:32  INFO: [F3DB58C0] id: 000027042385393, time: 2017-04-01 13:56:30, lat: -19.90913, lon: -43.97631, speed: 1.6, course: 229.9
2017-04-01 13:57:01  INFO: Position filtered by Distance filters from device: 000027042385393 with id: 1
2017-04-01 13:57:32  INFO: [F3DB58C0] id: 000027042385393, time: 2017-04-01 13:57:30, lat: -19.90888, lon: -43.97653, speed: 0.5, course: 214.6
2017-04-01 13:58:01  INFO: [F3DB58C0] id: 000027042385393, time: 2017-04-01 13:58:00, lat: -19.90914, lon: -43.97604, speed: 5.4, course: 129.2
2017-04-01 16:23:45  INFO: Position filtered by Duplicate Distance filters from device: 000027042385393 with id: 1
2017-04-01 16:27:50  INFO: Automatically registered device 027042385393
2017-04-01 16:27:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:27:47, lat: -19.91510, lon: -43.97345, speed: 5.1, course: 275.5
2017-04-01 16:27:52  INFO: Position filtered by Distance filters from device: 027042385393 with id: 20
2017-04-01 16:28:19  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:28:17, lat: -19.91505, lon: -43.97529, speed: 14.2, course: 269.0
2017-04-01 16:28:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:28:47, lat: -19.91722, lon: -43.97556, speed: 20.0, course: 183.8
2017-04-01 16:29:19  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:29:17, lat: -19.91966, lon: -43.97568, speed: 20.2, course: 184.7
2017-04-01 16:29:49  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:29:48, lat: -19.92132, lon: -43.97571, speed: 13.9, course: 191.4
2017-04-01 16:30:19  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:30:18, lat: -19.92316, lon: -43.97568, speed: 5.6, course: 167.9
2017-04-01 16:30:55  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:30:48, lat: -19.92325, lon: -43.97979, speed: 33.5, course: 285.1
2017-04-01 16:31:20  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:31:18, lat: -19.92239, lon: -43.98333, speed: 21.7, course: 267.0
2017-04-01 16:31:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:31:48, lat: -19.92426, lon: -43.98519, speed: 10.3, course: 133.4
2017-04-01 16:32:20  INFO: Position filtered by Distance filters from device: 027042385393 with id: 20
2017-04-01 16:32:51  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:32:48, lat: -19.92526, lon: -43.98450, speed: 24.5, course: 174.2
2017-04-01 16:33:20  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:33:19, lat: -19.92880, lon: -43.98339, speed: 31.2, course: 194.2
2017-04-01 16:33:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:33:49, lat: -19.93481, lon: -43.98300, speed: 46.8, course: 196.3
2017-04-01 16:34:20  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:34:19, lat: -19.94000, lon: -43.98566, speed: 22.9, course: 198.9
2017-04-01 16:34:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:34:49, lat: -19.94074, lon: -43.98654, speed: 30.7, course: 232.1
2017-04-01 16:35:20  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:35:19, lat: -19.94494, lon: -43.98699, speed: 37.0, course: 178.0
2017-04-01 16:35:50  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:35:49, lat: -19.94857, lon: -43.98521, speed: 40.6, course: 111.9
2017-04-01 16:36:21  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:36:19, lat: -19.95222, lon: -43.98390, speed: 39.6, course: 239.4
2017-04-01 16:36:51  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:36:50, lat: -19.95396, lon: -43.98579, speed: 0.9, course: 196.4
2017-04-01 16:37:21  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:37:20, lat: -19.95461, lon: -43.98600, speed: 23.1, course: 196.1
2017-04-01 16:37:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:37:50, lat: -19.95549, lon: -43.98642, speed: 0.0, course: 176.4
2017-04-01 16:38:22  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:38:20, lat: -19.95648, lon: -43.98619, speed: 15.1, course: 168.6
2017-04-01 16:38:53  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:38:50, lat: -19.95962, lon: -43.98451, speed: 28.3, course: 111.0
2017-04-01 16:39:21  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:39:20, lat: -19.96156, lon: -43.98223, speed: 16.7, course: 153.6
2017-04-01 16:39:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:39:50, lat: -19.96504, lon: -43.98217, speed: 21.8, course: 170.9
2017-04-01 16:40:22  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:40:20, lat: -19.96652, lon: -43.98161, speed: 13.2, course: 170.8
2017-04-01 16:40:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:40:50, lat: -19.96852, lon: -43.98111, speed: 10.9, course: 166.2
2017-04-01 16:41:22  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:41:21, lat: -19.97059, lon: -43.97990, speed: 11.8, course: 144.6
2017-04-01 16:41:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:41:51, lat: -19.97206, lon: -43.97850, speed: 10.6, course: 148.6
2017-04-01 16:42:22  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:42:21, lat: -19.97389, lon: -43.97959, speed: 10.7, course: 223.6
2017-04-01 16:42:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:42:51, lat: -19.97455, lon: -43.98027, speed: 11.8, course: 96.7
2017-04-01 16:43:22  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:43:21, lat: -19.97493, lon: -43.97981, speed: 9.8, course: 236.9
2017-04-01 16:43:52  INFO: [8F952F2E] id: 027042385393, time: 2017-04-01 16:43:51, lat: -19.97529, lon: -43.98007, speed: 0.4, course: 202.6
2017-04-01 16:44:23  INFO: Position filtered by Distance filters from device: 027042385393 with id: 20
2017-04-01 16:44:53  INFO: Position filtered by Distance filters from device: 027042385393 with id: 20
2017-04-01 17:58:20  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 16:44:51, lat: -19.97540, lon: -43.98007, speed: 0.4, course: 151.0
2017-04-01 17:59:30  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 17:59:29, lat: -19.97454, lon: -43.97681, speed: 7.8, course: 15.2
2017-04-01 18:00:00  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 17:59:58, lat: -19.97368, lon: -43.97653, speed: 17.8, course: 145.5
2017-04-01 18:00:30  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:00:29, lat: -19.97640, lon: -43.97758, speed: 20.1, course: 154.7
2017-04-01 18:01:00  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:00:59, lat: -19.97670, lon: -43.97482, speed: 15.7, course: 90.7
2017-04-01 18:01:35  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:01:29, lat: -19.97652, lon: -43.97356, speed: 16.2, course: 90.7
2017-04-01 18:02:01  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:01:59, lat: -19.97589, lon: -43.97083, speed: 10.6, course: 89.9
2017-04-01 18:02:31  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:02:29, lat: -19.97602, lon: -43.96959, speed: 8.4, course: 91.7
2017-04-01 18:03:01  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:02:59, lat: -19.97541, lon: -43.96865, speed: 11.6, course: 40.5
2017-04-01 18:03:31  INFO: [41D2652C] id: 000027042385393, time: 2017-04-01 18:03:30, lat: -19.97302, lon: -43.96779, speed: 14.0, course: 30.5
```

**Comment by tananaev** [comment]
Don't add both. Just add `027042385393` as an identifier.

**Comment by glbatistabh** [comment]
I did not add "consciously".
I have enabled the automatic registration in the configuration file.
```
    <entry key='database.registerUnknown'>true</entry>
    <entry key='database.registerUnknown.defaultGroupId'>1</entry>
```
Let me understand: if I add ONLY the code "027042385393" will the traccar "understand" both codes ("027042385393" and "000027042385393") as being the same device?

In this case, will I have to disable the "automatic registration" feature because the 2 options are conflicting?
